spirv_glsl.hpp

```diff
@@ -523,7 +527,10 @@ class CompilerGLSL : public Compiler
 	std::string to_unpacked_expression(uint32_t id, bool register_expression_read = true);
 	std::string to_unpacked_row_major_matrix_expression(uint32_t id);
 	std::string to_enclosed_unpacked_expression(uint32_t id, bool register_expression_read = true);
-	std::string to_dereferenced_expression(uint32_t id, bool register_expression_read = true);
+	
+    // Metal expands float[]/float2[] members inside structs to float4[] so we must unpack
```

Comment:
Formatting, there is a clang format file, FWIW.

Reply:
Was there a need to change this? Unpacking from float1/2 UBO array should already work.

Reply:
You might be right with that, too. I'm still doing internal testing which takes some time.